src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -986,25 +986,24 @@ internal async Task InvokeResponse(InvocationResponse invokeResponse)
                             }
                         }
 
-                        StringBuilder logBuilder = new StringBuilder();
-                        bool usedSharedMemory = false;
+                        StringBuilder sharedMemoryLogBuilder = null;
 
                         foreach (ParameterBinding binding in invokeResponse.OutputData)
                         {
                             switch (binding.RpcDataCase)
                             {
                                 case ParameterBindingType.RpcSharedMemory:
-                                    logBuilder.AppendFormat("{0}:{1},", binding.Name, binding.RpcSharedMemory.Count);
-                                    usedSharedMemory = true;
+                                    sharedMemoryLogBuilder ??= new StringBuilder();
+                                    sharedMemoryLogBuilder.AppendFormat("{0}:{1},", binding.Name, binding.RpcSharedMemory.Count);
                                     break;
                                 default:
                                     break;
                             }
                         }
 
-                        if (usedSharedMemory)
+                        if (sharedMemoryLogBuilder != null)
                         {
-                            _workerChannelLogger.LogDebug("Shared memory usage for response of invocation '{invocationId}' is {SharedMemoryUsage}", invokeResponse.InvocationId, logBuilder.ToString());
+                            _workerChannelLogger.LogDebug("Shared memory usage for response of invocation '{invocationId}' is {SharedMemoryUsage}", invokeResponse.InvocationId, sharedMemoryLogBuilder.ToString());
```

Comment:
nit: do not call `.ToString()`. Let the logger internals call `.ToString()`, this way if the log statement is filtered out we won't pay the price for that `ToString()` call at all.

Reply:
Good call. I didn't add that part with this PR, but I'll clean it up while I'm in here.